### PR TITLE
refactor(remix-dev): replace `esbuild-plugin-polyfill-node` with `esbuild-plugins-node-modules-polyfill`

### PR DESCRIPTION
**Pull request by LukeAskew**
`esbuild-plugin-polyfill-node` was introduced as part of https://github.com/remix-run/remix/pull/5274. Unfortunately, that plugin introduces some issues:

- It includes globals unnecessarily (#6380)
- Per [this comment](https://github.com/remix-run/remix/pull/5274#issuecomment-1537045614), it increases the build size due to the inclusion of globals.
-  There is a potential for conflict with other globals defined by downstream module code due to the strategy it uses for including globals. See attached screenshot.

Screenshot 2023-05-17 at 10 41 12 AM

This change replaces `esbuild-plugin-polyfill-node` with `esbuild-plugins-node-modules-polyfill`, which also uses jspm (via [modern-node-polyfills](https://github.com/Aslemammad/modern-node-polyfills)).

---

Closes: #6380

Testing Strategy: Replaced plugin in my remix app and did not encounter errors.

### CODE REVIEW

**Comment by ivogt** [comment]
I've noticed that the esbuild-plugin-polyfill-node also polyfills node:async_hooks etc (makes them unavailable with a stub unimplemented). Basically making `compatibility_flags = [ "nodejs_compat" ]` unusable. 

Is this PR handling these correctly and we can use nodejs_compat?

**Comment by LukeAskew** [comment]
@ivogt If I understand correctly this should only affect browser code so you should be able to use `node:async_hooks` in Cloudflare's compat mode since it's Node.js. I'm not sure how you'd implement async_hooks for browser code. (https://github.com/jspm/jspm-core/issues/28)

**Comment by ivogt** [comment]
@LukeAskew  Sorry, let me clarify. So, for Cloudflare Workers the build is handled as 'neutral'. Since workers support [nodejs_compat](https://developers.cloudflare.com/workers/runtime-apis/nodejs/) and there are couple of modules with node: prefix (e.g. node:utils and node:async_await) which these polyfills override. In https://github.com/remix-run/remix/pull/6414/files#diff-833c098a2a1bb29fb7094fb9a15af9f299cadbd1fb35bcc0cd7f078807fa8358 we remove the imports to anything `node:modules` which prevents us from using them. 

I suggest we exclude the workers from the polyfill e.g. `ctx.config.serverPlatform !== "node" && ctx.config.serverPlatform !== "neutral" `.

Don't think we need to polyfill for workers runtime, do we?

Thank you

**Comment by ivogt** [comment]
Also mentioned here https://github.com/remix-run/remix/issues/6348

File: packages/remix-dev/compiler/server/compiler.ts [hunk 1]
```diff
@@ -61,7 +61,7 @@ const createEsbuildConfig = (
   ];
 
   if (ctx.config.serverPlatform !== "node") {
```
Comment (ivogt) [inline packages/remix-dev/compiler/server/compiler.ts:63]:
```suggestion
  if (ctx.config.serverPlatform !== "node" && ctx.config.serverPlatform !== "neutral"  ) {
```

**Comment by bryce-pearce** [comment]
this would also likely close https://github.com/remix-run/remix/pull/6259

**Comment by frandiox** [comment]
Thanks for the PR. This should also close https://github.com/remix-run/remix/issues/6348 thanks to the [last commit](https://github.com/remix-run/remix/pull/6414/commits/8fe00b9c8b2351d67dc379e2c14b51d5a046ac2a).

**Comment by DaniBitZesty** [comment]
Hello, I'm working on a project that's quite dependent on this fix. Will this be merged soon? 🙏

**Comment by BrycePearce** [comment]
> Hello, I'm working on a project that's quite dependent on this fix. Will this be merged soon? 🙏

likewise

**Comment by MichaelDeBoey** [comment]
Superseded by #6547

**Comment by jacob-ebey** [comment]
I have broken this PR up into two parts:
1) Replace the polyfill package: https://github.com/remix-run/remix/pull/6562
2) Add config option for if polyfills are used or not: https://github.com/remix-run/remix/pull/6547

We can't accept this as is as it changes existing behavior for users. Thank you for the contribution and pointing us at a better alternative :)

**Comment by jacob-ebey** [comment]
Part 1 merged: https://github.com/remix-run/remix/pull/6562

**Comment by brandonpittman** [comment]
> Testing Strategy: Replaced plugin in my remix app and did not encounter errors.

I did.

**Pull request closed without merging**